File: go-client-sdk/pkg/models/shared/codatdatacontractsdatasetsinvoicelineitem.go
package shared

type CodatDataContractsDatasetsInvoiceLineItem struct {
	AccountRef           *CodatDataContractsDatasetsAccountRef                 `json:"accountRef,omitempty"`
	Description          *string                                               `json:"description,omitempty"`
	DiscountAmount       *float64                                              `json:"discountAmount,omitempty"`
	DiscountPercentage   *float64                                              `json:"discountPercentage,omitempty"`
	IsDirectIncome       *bool                                                 `json:"isDirectIncome,omitempty"`
	ItemRef              *CodatDataContractsDatasetsItemRef                    `json:"itemRef,omitempty"`
	Quantity             float64                                               `json:"quantity"`
	SubTotal             *float64                                              `json:"subTotal,omitempty"`
	TaxAmount            *float64                                              `json:"taxAmount,omitempty"`
	TaxRateRef           *CodatDataContractsDatasetsTaxRateRef                 `json:"taxRateRef,omitempty"`
	TotalAmount          *float64                                              `json:"totalAmount,omitempty"`
	Tracking             *CodatDataContractsDatasetsAccountsReceivableTracking `json:"tracking,omitempty"`
	TrackingCategoryRefs []CodatDataContractsDatasetsTrackingCategoryRef       `json:"trackingCategoryRefs,omitempty"`
	UnitAmount           float64                                               `json:"unitAmount"`
}

File: typescript-client-sdk/src/sdk/models/shared/codatdatacontractsdatasetsinvoicelineitem.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { CodatDataContractsDatasetsAccountRef } from "./codatdatacontractsdatasetsaccountref";
import { CodatDataContractsDatasetsAccountsReceivableTracking } from "./codatdatacontractsdatasetsaccountsreceivabletracking";
import { CodatDataContractsDatasetsItemRef } from "./codatdatacontractsdatasetsitemref";
import { CodatDataContractsDatasetsTaxRateRef } from "./codatdatacontractsdatasetstaxrateref";
import { CodatDataContractsDatasetsTrackingCategoryRef } from "./codatdatacontractsdatasetstrackingcategoryref";
import { Expose, Type } from "class-transformer";


export class CodatDataContractsDatasetsInvoiceLineItem extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "accountRef" })
  @Type(() => CodatDataContractsDatasetsAccountRef)
  accountRef?: CodatDataContractsDatasetsAccountRef;

  @SpeakeasyMetadata()
  @Expose({ name: "description" })
  description?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "discountAmount" })
  discountAmount?: number;

  @SpeakeasyMetadata()
  @Expose({ name: "discountPercentage" })
  discountPercentage?: number;

  @SpeakeasyMetadata()
  @Expose({ name: "isDirectIncome" })
  isDirectIncome?: boolean;

  @SpeakeasyMetadata()
  @Expose({ name: "itemRef" })
  @Type(() => CodatDataContractsDatasetsItemRef)
  itemRef?: CodatDataContractsDatasetsItemRef;

  @SpeakeasyMetadata()
  @Expose({ name: "quantity" })
  quantity: number;

  @SpeakeasyMetadata()
  @Expose({ name: "subTotal" })
  subTotal?: number;

  @SpeakeasyMetadata()
  @Expose({ name: "taxAmount" })
  taxAmount?: number;

  @SpeakeasyMetadata()
  @Expose({ name: "taxRateRef" })
  @Type(() => CodatDataContractsDatasetsTaxRateRef)
  taxRateRef?: CodatDataContractsDatasetsTaxRateRef;

  @SpeakeasyMetadata()
  @Expose({ name: "totalAmount" })
  totalAmount?: number;

  @SpeakeasyMetadata()
  @Expose({ name: "tracking" })
  @Type(() => CodatDataContractsDatasetsAccountsReceivableTracking)
  tracking?: CodatDataContractsDatasetsAccountsReceivableTracking;

  @SpeakeasyMetadata({ elemType: CodatDataContractsDatasetsTrackingCategoryRef })
  @Expose({ name: "trackingCategoryRefs" })
  @Type(() => CodatDataContractsDatasetsTrackingCategoryRef)
  trackingCategoryRefs?: CodatDataContractsDatasetsTrackingCategoryRef[];

  @SpeakeasyMetadata()
  @Expose({ name: "unitAmount" })
  unitAmount: number;
}